neurons/validator.py

```diff
@@ -635,6 +676,124 @@ def get_positions(self, synapse: template.protocol.GetPositions,
         synapse.error_message = error_message
         return synapse
 
+    # Enable validator <-> validator p2p checkpoint sync
+    def checkpoint_thread(self):
```

Comment:
we don't want a thread for this. It should be run in the validator main loop so it is synchronized with the other threads. This is because we don't want to to introduce a race condition where our new positions are overwritten by another thread.

We should follow what autosync in currently doing